language/bytecode-verifier/src/locals_safety/abstract_state.rs

```diff
@@ -44,15 +46,16 @@ impl AbstractState {
                     .locals,
             )
             .0;
-        let func_handle = module.function_handle_at(function_definition.function);
+        let num_locals = parameter_types.len() + additional_local_types.len();
 
         let num_args = module.signature_at(func_handle.parameters).0.len();
-        let local_states = (0..local_types.len())
+        let local_states = (0..num_locals)
             .map(|i| if i < num_args { Available } else { Unavailable })
             .collect();
 
-        let local_kinds = local_types
+        let local_kinds = parameter_types
             .iter()
+            .chain(additional_local_types.iter())
```

Comment:
nit remove the `.iter()` and add an `&` if needed

Reply:
BTW just curious, is there a reason you strongly prefer using `&` instead of `.iter()` for chain? I'm fine with both, but semantically `.iter()` seems more natural to me as it clearly states it's the concatenation of two iterators, while the & will go through an `into_iter()` conversion. (The doc for chain says it "Takes two iterators and creates a new iterator over both in sequence.", and the example for `zip` uses `iter().zip(other.iter())`)

Reply:
I like it because it provides better symmetry between cases

```
// owned case
x.into_iter().zip(y)
// ref case
x.iter().zip(&y)
```

I like both of these better than 
```
x.into_iter().zip(y.into_iter())
x.iter().zip(y.iter())
```
strictly because If ind them easier to read. 

Again its just a nit, so you do you if you don't like it :)

Reply:
In short, pick a style and stick with it. If you always call `.iter()` and `.into_iter()`, do that
If you always do `&` and no call, do that

Mixing some cases where it is `.iter()` and then no call for the  `.into_iter()` case feels weird to me

Reply:
> while the & will go through an into_iter() conversion. 

The same thing is happening fwiw in the `.iter()` case. Almost exactly the same set of calls

Reply:
> The same thing is happening fwiw in the .iter() case. Almost exactly the same set of calls
Fully agreed, although I'd say semantically it's a no-op. The way I see it is that, from the callee's perspective, it is trying to generalize the concept by doing an extra step to take not just iterators, but things that can be converted into iterators, giving more choices to the caller. To me as a caller, I kinda want to be explicit about my intention -- which is to chain two iterators and writing `.iter()` makes me feel that I'm constraining the type as much as I can. (But I agree it's a bit more verbose than the other style.)